src/Uno.UI/UI/Xaml/Media/ImageBrush.Android.cs

```diff
@@ -25,12 +25,37 @@ public partial class ImageBrush
 		private bool _imageSourceChanged;
 		private Windows.Foundation.Rect _lastDrawRect;
 		private readonly SerialDisposable _refreshPaint = new SerialDisposable();
+		private readonly SerialDisposable _sourceDisposable = new SerialDisposable();
+
+		internal event Action ImageChanged;
 
 		partial void OnSourceChangedPartial(ImageSource newValue, ImageSource oldValue)
 		{
+
+
+			if (newValue is not null)
+			{
+				newValue.InvalidateSource += OnInvalidateSource;
+				_sourceDisposable.Disposable = Disposable.Create(() => newValue.InvalidateSource -= OnInvalidateSource);
+
+				void OnInvalidateSource(object sdn, EventArgs args)
+				{
+					oldValue?.Dispose();
```

Comment:
Hummm what happen if we use the `oldValue` on multiple `Image` or `ImageBrush`, or if we re-use it later ? I don't think that we should dispose it here.